drivers/video/ov5640.c

```diff
@@ -457,23 +635,84 @@ static int ov5640_write_multi_regs(const struct i2c_dt_spec *spec, const struct
 	return 0;
 }
 
+static int ov5640_set_fmt_dvp(const struct ov5640_config *cfg)
+{
+	uint8_t reg;
+	int ret = 0;
+
+	ret |= ov5640_read_reg(&cfg->i2c, TIMING_TC_REG21_REG, &reg, sizeof(reg));
+	ret |= ov5640_write_reg(&cfg->i2c, TIMING_TC_REG21_REG, (reg & 0xDF) | 0x00);
+
+	if (ret) {
+		LOG_ERR("Unable to configure REG: %d on DVP", TIMING_TC_REG21_REG);
+		return ret;
```

Comment:
If `read()` and `write()` have different errors, then `ret` will not contain a valid error code, so cannot be returned directly... I am still unsure of the best way to handle this while avoiding one error message per line. Maybe this?

```
if ((ret = ov5640_read_reg(&cfg->i2c, TIMING_TC_REG21_REG, &reg, sizeof(reg))) ||
    (ret = ov5640_write_reg(&cfg->i2c, TIMING_TC_REG21_REG, (reg & 0xDF) | 0x00))) {
        LOG_ERR(...);
        return ret;
}
```

Although this style is also rare in Zephyr.

In this particular case, though, this sounds like solved by `ov5640_modify_reg()`.

Reply:
As seen below, you can also use a const table (i.e. global at the top) of addresses/value/mask to apply.

Reply:
> In this particular case, though, this sounds like solved by `ov5640_modify_reg()`.

`ov5640_modify_reg()` sounds great! I'll change to use it.

Reply:
Looking again at various image sensor drivers, the `ret |= ...` seems present in various sensor.
Contributions tend to preserve it, but do feel free to break these local changes for the more Zephyr-like `ret = ...; if (ret) { return ret; }` for new code. :)